core/src/thread_parker/wasm.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright 2016 Amanieu d'Antras
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+//! A simple spin lock based thread parker. Used on platforms without better
+//! parking facilities available.
```

Comment:
I think this should be changed.

Reply:
Yes it should. Thank you. I just started out with the generic impl and iterated from there. Forgot some stuff :P